### PR TITLE
TRC-40- Add `TrendingItemSuggestion` Component

### PR DESCRIPTION
**Pull request by mmcbride2929**
## TL/DR
This story involves creating a `TrendingItemSuggestion` component that displays suggested reddit posts.

TrendingItemSuggestion

## Overview of Changes

`TrendingItemSuggestion.tsx`
- Built an interface that accepts a metaData object, a function & an optional customTestId
- Destructures metaData object
- Conditionally renders the icon depending on if metaData has been fetched

`TrendingItemSuggestion.stories.tsx`
- A Storybook story that displays our component, and has the metaData args so we can preview the component with data.

`TrendingItemSuggestion.scss`
- Stylesheet for our component + it's hover state

`TrendingItemSuggestion.test.tsx`
- 7 tests to make sure our elements render properly
- 1 test to determine our handleClick function fires onClick

### CODE REVIEW

**Comment by wijohnst** [comment]
I'm noticing that we never defined a body font in the global CSS. I'll add a story to the project so we can get that added.

File: apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    trendingSuggestionHeading,
+    trendingSuggestionSummary,
+    targetSubreddit,
+  } = metaData;
+  const { subRedditIcon, subRedditName } = targetSubreddit;
+
+  return (
+    <li
+      onClick={handleClick}
+      className={!metaData ? 'loading' : ''}
```
Comment (wijohnst) [inline apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx:36]: I'm not sure this will work the way that you think it will. `metaData` is a required prop, so this will always be truthy. We are going to be using `ReactQuery` for out FE data fetching / caching and that will expose a the async lifecycle values that can be used to handle an `isLoading` prop. This prop can be consumed by any component that is dependent on async data. 

We will do the actual data fetching in `TrendingItemSuggestion`'s parent component - `TrendingSuggestionList`. It will look something like this:

```TrendingSuggestionList.tsx
import { useQuery } from 'react-query'
import { fetchTrendingSuggestions } from './TrendingSuggestions.api.ts'

// fetchTrendingSuggestions is an async function (probably Axios) that makes the actual request to the endpoint
// useQuery is a custom hook exposed by `react-query`
// data is the same as `typeof fetchTrendingSuggestions`, in this case an array of `TrendingSuggestionItems`, // which should have the same shape as the `metaData` object
const { isFetching, isFetched, data } =  useQuery('trendingSuggestions', fetchTrendingSuggestions); 

return(
{data.map((trendingSuggestionData: TrendingSuggestionData)) => (
  <TrendingItemSuggestion 
    isLoading={!isFetched || isFetching}
    ...trendingSuggestionData
  />
)}
)
```

`react-query` provides us with a wrapper for fetching our data as well as an API for dealing with the async lifecycle all in a very declarative way.

Comment (wijohnst) [reply apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.tsx:36]: @willcary Want to make sure you see this comment.

**Review by wijohnst — COMMENTED** [review]
Great start.

File: apps/web/src/components/TrendingItemSuggestion/TISmetaData.tsx
Comment (wijohnst) [inline apps/web/src/components/TrendingItemSuggestion/TISmetaData.tsx:1]: Rename to `TrendingItemSuggestion.utils.tsx`.

File: apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const Template: ComponentStory<typeof TrendingItemSuggestion> = (args) => (
+  <TrendingItemSuggestion {...args} />
+);
+
+export const Default = Template.bind({});
+Default.args = { metaData: metaData, isLoading: false };
+
+export const Loading = Template.bind({});
+Loading.args = { metaData: metaData, isLoading: true };
```
Comment (wijohnst) [inline apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.stories.tsx:25]: Remember to use the spread operator on your `Default.arg`. I find this to be the most flexible and consistent pattern. 


```suggestion
Loading.args = { ...Default.args, isLoading: true };
```

**Review by wijohnst — COMMENTED** [review]
Left a couple more small changes.

**Review by wijohnst — APPROVED** [review]
LGTM.
